### DISCUSSION TITLE
.get_decomposition_model() for blind source separation

### DISCUSSION THREAD

**Discussion by lscaldas** [hyperspy/hyperspy#3378] [Q&A] [answered]
Hello, is there a way (native or not) to reconstruct the signal without a component found in the blind_source_separation method? Similarly to how you would do it with the decomposition for denoising?

**Comment by francisco-dlp** [comment, marked as answer]
Just pass the components you want in a list e.g.:

```python
sbss = s.get_bss_model(components=[0,2])
```

**Reply by lscaldas** [reply]
Thank you Francisco. This works. But I am still having another issue preventing me to reach my goal, that maybe you can help. 

My data/ signal is originally like this: <Signal2D, title: , dimensions: (133|600, 400)>   and then I transpose with  signal.transpose(signal_axes=[0]), making it <Signal1D, title: , dimensions: (600, 400|133)>, 

I then proceed to do decomposition, and then bss. I can use the plot_bss_results() function to see the result (correct result), and I also can use the get_bss_model as you suggested. However, I cant use the save method successfully, with the TypeError: 

This file format does not support this data. Please try one of Electron Microscopy Data (EMD), HSPY, Nexus, Ripple, SEMPER and ZSpy .

I can only use the save after the get_bss_model, if I dont originally transpose the data. But then the components are not correct. 

My data originally is a tiff image with x,y dimensions of 600,400, with the 133 as the energy dimension (133 images). I am using hyperspy 1.7.5.

Do you know a way of bypassing this issue, exporting or saving the data (reconstruction without a specific bss component) from a signal 1d bss operation?

**Reply by francisco-dlp** [reply]
The issue is that you cannot save spectrum images to TIFF. You can workaround the issue by transposing back your data. For example:

```python
im = hs.load("original_data.tiff")
s = im.T
s.decomposition()
s.blind_source_separation(3)
sbss = s.get_bss_model([0,2])
imbss = sbss.T
imbss.save("bss_data.tiff")
```

**Reply by lscaldas** [reply]
that worked like a charm! Thank you very much!!
